applications/SEOR-code-less-smart-contract-platform.md

```diff
@@ -0,0 +1,155 @@
+| ------------- | ------------- | ------------- |
+| 1. | License | Apache License 2.0 |
+| 2. | Contract set Repository | A git repository containing the source code of 4 contracts: parameterized ERC20, liquidity mining, contract deployer and Uniswap protocol connector. These smart contracts are all developed to realizes parameterized during deployment, not copy the standard contract such as ERC20 directly. All of contracts will be will be implemented using both solidity and ink! technologies. |
+| 3. | Testing Guide | All contracts will have a proper unit-test coverage (min 80%) to ensure the robustness and safety of the contract. All of the test cases, test codes and test guide will be add to the repository. |
+| 4. | Documentation | Related documents such as interface definitions, usage instructions, deployment tutorials, etc. of the contract set. These documents will be added to the git repository, and a pdf file download link will be provided on the project website. |
+
+### Milestone 2 - Multi-chain SERO-JS-SDK
+* **Estimated Duration:** 1 month
+* **FTE:**  3
+* **Costs:** 5000 DAI
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | License | Apache License 2.0 |
+| 2. | SDK code repository | A git repository of SEOR-JS-SDK implemented using TypeScript. The SDK uses a unified interface to support the interaction between front-end DApp and substrate nodes, Ethereum nodes and other blockchain nodes of different architectures, allowing developers to access different chains with minimal changes without modifying the logic.  |
+| 3. | Testing Guide | The SDK with have enough unit-test coverage(min 80%) to ensure  functional integrity and robustness. All of the test cases, test codes and test guides will be added to the repository. |
+| 4. | Documentation | The SERO-JS-SDK tutorial and documentation guide users how to quickly get started using it, and provide as detailed use cases as possible. These documents will be added to the git repository, and a pdf file download link will be provided on the project website. |
+
+
+### Milestone 3 - Multi-chain codeless smart contract platform
+* **Estimated Duration:** 1 month
+* **FTE:**  2
+* **Costs:** 2500 DAI
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | License | Apache License 2.0 |
+| 2. | Code-less smart contractPlatform DApp | A code-less smart contract platform DApp will be deployed under a second-level domain name of this project. The DApp's UI part will be implement by VUE, and it will have a custom designed UI to uniform different chain's interface. The interactive with these chains will be supported by SEOR-JS-SDK. Through this DApp, users can deploy the contract implemented in Milestone 1 without coding to the parachains supporting smart contracts or directly to Ethereum, and contract interaction can be carried out in the platform powered by SEOR-JS-SDK. |
```

Comment:
Deploying smart contracts to public chains involves paying a Tx fee. Unless you're going to sponsor such deployments, users will need a way to connect to a wallet (multiple wallets if deploying to multiple chains?) - how do you envision this integration?

Reply:
We have no plans to pay Tx fees for users, so users must using different wallet  to pay the fee for different chains. 
We will support multiple wallets through a set of unified interface SDK wich implemented in Mile stone-2 and handle the interaction with the chain.  

In fact, we have previously developed a set of unified interface SDK, which supports wallet protocols of different public chains such as ETH, EOS, and ONT. However, due to various reasons, this SDK has not been maintained continuously.

This is the previously SDK documents:
[protocol](https://sealsc.github.io/documents/en/web-extensions/protocol/)
[proxy](https://sealsc.github.io/documents/en/web-extensions/proxy/)
[metamask-extension](https://sealsc.github.io/documents/en/web-extensions/metamask/)
[scatter-extension](https://sealsc.github.io/documents/en/web-extensions/scatter/)
[cyano-extension](https://sealsc.github.io/documents/en/web-extensions/cyano/)

Based on this SDK, we will use TypeScript to rewrite and improve this SDK in the Milestone-2 of this application to support more wallets.

Reply:
> Based on this SDK, we will use TypeScript to rewrite and improve this SDK in the Milestone-2 of this application to support more wallets.

Could you update the application document and include this? Please list the wallets that your SDK would support (at least one should be a Polkadot wallet)

Reply:
@mmagician Thanks for the review!

The deliverables have been updated.

We add the multi-chain & wallets support in milestone 2 and milestone 3's deliverable, and list all the chains and wallets we will be supported on this application.

Reply:
Great!